### PR TITLE
Add docker dev setup and update requirements

### PR DESCRIPTION
**Pull request by thecharlieblake**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -15,17 +15,19 @@ jobs:
...
-        run: ./dev ci
+        run: docker run --rm -v $(pwd):/workspace unit-scaling-dev:latest ./dev ci
+
       - name: Publish documentation
         if: ${{github.ref == 'refs/heads/main'}}
         uses: Cecilapp/GitHub-Pages-deploy@v3
         env: { GITHUB_TOKEN: "${{ github.token }}" }
         with:
-          build_dir: docs/_build/html
+          build_dir: docker run --rm -v $(pwd):/workspace unit-scaling-dev:latest docs/_build/html
```
Comment (DouglasOrr) [inline .github/workflows/ci.yaml:33]: Is this right - I thought it looked like a dir not a command?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+RUN groupadd --gid $USER_GID $USERNAME \
+    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME \
+    && echo "$USERNAME ALL=(root) NOPASSWD:ALL" > /etc/sudoers.d/$USERNAME \
+    && chmod 0440 /etc/sudoers.d/$USERNAME
+USER $USERNAME
+
+# Creates basic .zshrc
+RUN sudo cp /etc/zsh/newuser.zshrc.recommended /home/$USERNAME/.zshrc
+
+CMD ["/bin/zsh"]
```
Comment (DouglasOrr) [inline Dockerfile:39]: Why zsh, out of interest? Just personal preference? (No objection, just curious.)

**Review by DouglasOrr — APPROVED** [review]
LGTM; thanks!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (thecharlieblake) [reply .github/workflows/ci.yaml:33]: It's completely mad! No idea what's going on there. Removed

**Pull request merged**